zerver/views/webhooks/__init__.py

```diff
@@ -0,0 +1,49 @@
+from django.conf.urls import url
+
+
+class Integration(object):
+    DEFAULT_LOGO_STATIC_PATH = 'static/images/integrations/logo/{name}'
+    DEFAULT_FUNCTION_PATH = 'zerver.views.webhooks.{name}.api_{name}_webhook'
+    DEFAULT_CLIENT_NAME = 'Zulip{name}Webhook'
+    DEFAULT_URL = 'api/v1/external/{name}'
+
+    def __init__(self, name, logo=None, function=None, client_name=None, url=None):
+        self.name = name
+        self.logo = logo if logo else self.DEFAULT_LOGO_STATIC_PATH.format(name=name)
+        self.function = function if function else self.DEFAULT_FUNCTION_PATH.format(name=name)
+        self.client_name = client_name if client_name else self.DEFAULT_CLIENT_NAME.format(name=name.title())
+        self.url = url if url else self.DEFAULT_URL.format(name=name)
```

Comment:
Can you write these instead in the style of
```
if logo is None:
    logo = self.DEFAULT_LOGO_STATIC_PATH.format(name=name)
self.logo = logo
```
While it is less compact code, it's easier to read, and I think it will age better as we add into this list non-webhook integrations (where there is no `function`, for example).